components/radial-chart/src/radial-chart.ts

```diff
@@ -0,0 +1,293 @@
+import {
+  AfterContentInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChildren,
+  ElementRef,
+  Input,
+  OnDestroy,
+  QueryList,
+  ViewEncapsulation,
+  Directive,
+  TemplateRef,
+  ContentChild,
+} from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+import { DT_CHART_COLOR_PALETTE_ORDERED } from '@dynatrace/barista-components/chart';
+import {
+  DtViewportResizer,
+  isNumber,
+} from '@dynatrace/barista-components/core';
+import { PieArcDatum } from 'd3-shape';
+import { combineLatest, of, Subject } from 'rxjs';
+import { switchMap, takeUntil, startWith } from 'rxjs/operators';
+import { DtRadialChartSeries } from './radial-chart-series';
+import { DtRadialChartRenderData } from './utils/radial-chart-interfaces';
+import {
+  generatePathData,
+  generatePieArcData,
+  getSum,
+} from './utils/radial-chart-utils';
+import { DtRadialChartOverlayData } from './radial-chart-path';
+
+/** Size of the inner (empty) circle in proportion to the circle's radius. */
+const DONUT_INNER_CIRCLE_FRACTION = 0.8;
+
+@Directive({
+  selector: '[dtRadialChartOverlay]',
+})
+export class DtRadialChartOverlay {}
+
+@Component({
+  selector: 'dt-radial-chart',
+  exportAs: 'dtRadialChart',
+  templateUrl: 'radial-chart.html',
+  styleUrls: ['radial-chart.scss'],
+  host: {
+    class: 'dt-radial-chart',
+    '[class.dt-radial-chart-legend-right]': 'legendPosition === "right"',
+    '[class.dt-radial-chart-legend-bottom]': 'legendPosition === "bottom"',
+  },
+  preserveWhitespaces: false,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.Emulated,
+})
+export class DtRadialChart implements AfterContentInit, OnDestroy {
+  /**
+   * The chart type. Can either be 'pie' or 'donut'.
+   * Defaults to 'pie'.
+   */
+  @Input()
+  get type(): 'pie' | 'donut' {
+    return this._type;
+  }
+  set type(value: 'pie' | 'donut') {
+    if (value && value !== this._type) {
+      this._type = value;
+      this._updateRenderData();
+    }
+  }
+  private _type: 'pie' | 'donut' = 'pie';
+
+  /** Maximum value the chart displays, 100%, full circle */
+  @Input()
+  get maxValue(): number | null {
+    return this._maxValue;
+  }
+  set maxValue(value: number | null) {
+    this._maxValue = coerceNumberProperty(value);
+    this._updateRenderData();
+  }
+  private _maxValue: number | null = null;
+
+  /** Where the chart's legend should be placed */
+  @Input()
+  get legendPosition(): 'right' | 'bottom' {
+    return this._legendPosition;
+  }
+  set legendPosition(value: 'right' | 'bottom') {
+    this._legendPosition = value;
+  }
+  private _legendPosition: 'right' | 'bottom' = 'right';
+
+  /** @internal Series data, <dt-radial-chart-series> */
+  @ContentChildren(DtRadialChartSeries) _radialChartSeries: QueryList<
+    DtRadialChartSeries
+  >;
+
+  /** @internal Overlay template */
+  @ContentChild(DtRadialChartOverlay, { static: false, read: TemplateRef })
```

Comment:
`ContentChild` with static false is ok in this case, because we do not need to react, we can rely on the property being set (or not) when we open the overlay